### PR TITLE
ARROW-4362: [Java] Test OpenJDK 11 in CI

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
This PR fails for JDK 11 because of 

```
if [ $TRAVIS_OS_NAME == "linux" ]; then
    # Use the Ubuntu-provided OpenJDK
    unset JAVA_HOME
    export TRAVIS_MVN=/usr/bin/mvn
    export TRAVIS_JAVA=/usr/bin/java
else
    export TRAVIS_MVN=mvn
    export TRAVIS_JAVA=java
fi
```

Isn't this ignoring our JDK setup fully and simply reverting to the default JDK?

**Comment by xhochy** [comment]
The R failure is unrelated.

This PR also fixes the JDK 9 failure on master.

**Comment by praveenbingo** [comment]
Can you please rebase/retrigger tests, the http llvm issues are fixed now.

**Comment by xhochy** [comment]
@praveenbingo Rebased and all tests passed.

File: .travis.yml [hunk 1]
```diff
@@ -239,15 +240,19 @@ matrix:
   - name: "Java w/ Oracle JDK 9"
```
Comment (kou) [inline .travis.yml]: Should we update the name to "OpenJDK 9"?

Comment (wesm) [reply .travis.yml]: I just fixed

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**